src/radix/tree.cr

```diff
@@ -392,7 +392,7 @@ module Radix
         key_reader.next_char
       end
 
-      (!key_reader.has_next? && !different) &&
+      (!different) &&
```

Comment:
Definitely we should add a comment indicating what `_same_key?` does or how it reached to this result?